src/main/java/com/snowplowanalytics/redash/RedashClient.java

```diff
@@ -0,0 +1,437 @@
+    private static final String MEMBERS_URL_PREFIX = "/members";
+
+    private static final String ID = "id";
+    private static final String USER_ID = "user_id";
+    private static final String DATA_SOURCE_ID = "data_source_id";
+    private static final String AUTH_TYPE = "auth_type";
+    private static final String CREATED_AT = "created_at";
+    private static final String MESSAGE = "message";
+    private static final String NULL = "null";
+    private static final String MESSAGE_URL_NOT_FOUND = "The requested URL was not found on the server.  " +
+                                            "If you entered the URL manually please check your spelling and try again.";
+    private static final String MESSAGE_INTERNAL_SERVER_ERROR = "Internal Server Error";
+    private static final String JSON_CONTENT_TYPE = "application/json; charset=utf-8";
+
+    private static final MediaType JSON = MediaType.parse(JSON_CONTENT_TYPE);
+
+    public static final String DATA_SOURCE_ALREADY_EXISTS = "Data-source with this name already exists.";
+    public static final String USER_GROUP_ALREADY_EXISTS = "User group with this name already exists.";
+    public static final String USER_DOES_NOT_EXIST = "User with such name does not exist.";
+    public static final String DATA_SOURCE_DOES_NOT_EXIST = "Data-source with such name does not exist.";
+
+    private final OkHttpClient client;
+    private final String baseUrl;
+    private final String apiKey;
+    private final Headers headers;
+
+    public RedashClient(String schema, String host, int port, String apiKey) {
+        this.client = new OkHttpClient();
+        this.apiKey = apiKey;
+        this.baseUrl = schema + "://" + host + ":" + port + API_PREFIX;
+        this.headers = new Headers.Builder()
+                .add("Accept", "application/json, text/plain, */*")
+                .add("Content-Type", JSON_CONTENT_TYPE)
+                .build();
+    }
+
+    /**
+     * Creates new data-source instance on the Redash server.
+     *
+     * @param dataSource data transfer object which should contain all necessary information.
+     * @return int id of successfully (if so) created instance. In this case object that was transferred as argument
+     * receives that id.
+     * @throws IOException              if server is unable due to connection error or API key is invalid.
+     * @throws IllegalArgumentException if there is already presents data-source on Redash server
+     *                                  with such name
+     */
+    public int createDataSource(DataSource dataSource) throws IOException, IllegalArgumentException {
+        if (isEntityAlreadyExists(getDataSources(), dataSource.getName())) {
+            throw new IllegalArgumentException(DATA_SOURCE_ALREADY_EXISTS);
+        }
+        String url = baseUrl + DATA_SOURCES_URL_PREFIX + API_KEY_URL_PARAM + apiKey;
+        String response = post(url, new Gson().toJson(dataSource), CheckResponseStatus.NO);
```

Comment:
What happens here if the response status failed?  Will it fail with a NullPointer or Json exception?  Is there a reason not to catch the response and handle this failure state gracefully?

Reply:
It's my fail, of course we need to check the response. If something goes wrong then an IOException should be thrown.